ranndy/data_iterator.py

```diff
@@ -119,11 +118,3 @@ def load_dataset(self, shuffle):
         self.next_element = self.iterator.get_next()
         self.initializer = self.iterator.initializer
         self.source, self.target_input, self.target_output, self.source_length, self.target_length = self.next_element
-
-    def lookup_indexes(self, words):
-        """ Returns the index(es) for the given word(s) in the vocabulary """
-        return tf.to_int32(self.table.lookup(tf.convert_to_tensor(words)))
-
-    def lookup_words(self, indexes):
```

Comment:
I don't think these functions should be removed, as they are useful abstractions that make the code more readable (see above and in other file where these functions were removed). Maybe just add to the docstring that they add Tensors to the graph.

Reply:
It was a wrapper function to a tensorflow function. I don't see it adding enough value to be it's own function. All the extra function does it casting.

Reply:
Yes, but wrappers are good if it improves readability and reduces code duplication. See lines 38 and 39 (for lookup_word) in models.py and lines 70, 71 and 85 (for lookup_indexes). At the minimum I think we should keep lookup_words since "self.iterator.reverse_table.lookup(tf.to_int64(self.iterator.source))" is pretty confusing.

Reply:
Yes, but wrappers are good if it improves readability and reduces code duplication. See lines 38 and 39 (for lookup_word) in models.py and lines 70, 71 and 85 (for lookup_indexes). At the minimum I think we should keep lookup_words since "self.iterator.reverse_table.lookup(tf.to_int64(self.iterator.source))" is pretty confusing.